tests/test_single_topology_confgen.py

```diff
@@ -0,0 +1,100 @@
+import csv
+
+import numpy as np
+from rdkit import Chem
+
+from timemachine.constants import DEFAULT_FF, DEFAULT_TEMP
+from timemachine.fe import atom_mapping, pdb_writer, utils
+from timemachine.fe.rbfe import HostConfig, setup_initial_states
+from timemachine.fe.single_topology import AtomMapMixin, SingleTopology
+from timemachine.fe.utils import get_mol_name
+from timemachine.ff import Forcefield
+from timemachine.md import builders
+
+
+def get_mol_by_name(mols, name):
+    for m in mols:
+        if get_mol_name(m) == name:
+            return m
+    assert 0, "Mol not found"
+
+
+def write_trajectory_as_pdb(mol_a, mol_b, core, all_frames, host_topology, out_path):
+    atom_map_mixin = AtomMapMixin(mol_a, mol_b, core)
+    writer = pdb_writer.PDBWriter([host_topology, mol_a, mol_b], out_path)
+    for frame in all_frames:
+        host_frame = frame[: host_topology.getNumAtoms()]
+        ligand_frame = frame[host_topology.getNumAtoms() :]
+        mol_ab_frame = pdb_writer.convert_single_topology_mols(ligand_frame, atom_map_mixin)
+        writer.write_frame(np.concatenate([host_frame, mol_ab_frame]) * 10)
+    writer.close()
+
+
+def run_edge(mol_a, mol_b, protein_path, fname):
+
+    mcs_result = atom_mapping.mcs(mol_a, mol_b, conformer_aware=False)
+    threshold = 10.0
+    core = atom_mapping.get_core_by_mcs(mol_a, mol_b, mcs_result.queryMol, threshold)
+
+    res = utils.plot_atom_mapping_grid(mol_a, mol_b, mcs_result.smartsString, core)
+    with open(f"{fname}_map.svg", "w") as fh:
+        fh.write(res)
+
+    ff = Forcefield.load_from_file(DEFAULT_FF)
+    st = SingleTopology(mol_a, mol_b, core, ff)
+
+    lambda_schedule = np.linspace(0, 1, 32)
+    seed = 2023
+
+    # solvent
+    box_width = 4.0
+    solvent_sys, solvent_conf, solvent_box, solvent_top = builders.build_water_system(box_width, ff.water_ff)
+    solvent_box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+    solvent_host_config = HostConfig(solvent_sys, solvent_conf, solvent_box)
+    initial_states = setup_initial_states(st, solvent_host_config, DEFAULT_TEMP, lambda_schedule, seed)
+    all_frames = [state.x0 for state in initial_states]
+    write_trajectory_as_pdb(
+        mol_a,
+        mol_b,
+        core,
+        all_frames,
+        solvent_top,
+        f"solvent_{get_mol_name(mol_a)}_{get_mol_name(mol_b)}.pdb",
+    )
+
```

Comment:
Add some sanity check here (maybe rmsd to the mol_a/mol_b structures as appropriate?)

Reply:
1) the internal BFGS code already checks that the energy decreases
2) i can add a check that the drift relative to the starting configuration is small

Reply:
I added a check that so that no atom drifts more than 7 angstroms from the initial state. (arbitrary)

This is now done directly in the setup code.